docs/tutorials/demo_data_science_project.py

```diff
@@ -389,6 +389,15 @@ def transform_json_weather_data(filepath):
 Tags are immutable once created, so attempting to tag two different commits with the same name will result in an error.
 """
 
+# %% tags=["Remove_input"]
+# Remove tag "train-test-split-2010" if it exists, ensuring idempotency
+# Cell hidden in docs
+from lakefs_spec.client_helpers import list_tags, delete_tag
+
+if any(tag.id == "train-test-split-2010" for tag in list_tags(fs.client, REPO_NAME)):
+    delete_tag(fs.client, repository=REPO_NAME, tag="train-test-split-2010")   
```

Comment:
I checked; the commit in question is for the train/test split on the `training` branch.

Reply:
Even though, you did not mention repo deletion, just a note on it: That is not possible since the storage namespace is not cleared. 

However, I like the idea and will move the cell to the end of the notebook and clean up the branches.